### PR TITLE
feat(components): [menu] add `expand-close-icon`、`expand-open-icon`、`collapse-close-icon`、`collapse-open-icon`  props

### LINKED ISSUE #9425: [Component] [menu] menu展开、收起图标可自定义
希望自定menu展开、收起图标可自定义，满足如下场景
图片

### PR DESCRIPTION
**Pull request by xiaoxian521**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by xiaoxian521** [comment]
[example](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtcmFkaW8tZ3JvdXAgdi1tb2RlbD1cImlzQ29sbGFwc2VcIiBzdHlsZT1cIm1hcmdpbi1ib3R0b206IDIwcHhcIj5cbiAgICA8ZWwtcmFkaW8tYnV0dG9uIDpsYWJlbD1cImZhbHNlXCI+ZXhwYW5kPC9lbC1yYWRpby1idXR0b24+XG4gICAgPGVsLXJhZGlvLWJ1dHRvbiA6bGFiZWw9XCJ0cnVlXCI+Y29sbGFwc2U8L2VsLXJhZGlvLWJ1dHRvbj5cbiAgPC9lbC1yYWRpby1ncm91cD5cbiAgPGVsLW1lbnUgZGVmYXVsdC1hY3RpdmU9XCIyXCIgY2xhc3M9XCJlbC1tZW51LXZlcnRpY2FsLWRlbW9cIiA6Y29sbGFwc2U9XCJpc0NvbGxhcHNlXCIgQG9wZW49XCJoYW5kbGVPcGVuXCJcbiAgICBAY2xvc2U9XCJoYW5kbGVDbG9zZVwiPlxuICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjFcIiA6ZXhwYW5kLWNsb3NlLWljb249XCJDYXJldEJvdHRvbVwiIDpleHBhbmQtb3Blbi1pY29uPVwiQ2FyZXRUb3BcIj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+XG4gICAgICAgIDxlbC1pY29uPlxuICAgICAgICAgIDxsb2NhdGlvbiAvPlxuICAgICAgICA8L2VsLWljb24+XG4gICAgICAgIDxzcGFuPk5hdmlnYXRvciBPbmU8L3NwYW4+XG4gICAgICA8L3RlbXBsYXRlPlxuICAgICAgPGVsLW1lbnUtaXRlbS1ncm91cD5cbiAgICAgICAgPHRlbXBsYXRlICN0aXRsZT48c3Bhbj5Hcm91cCBPbmU8L3NwYW4+PC90ZW1wbGF0ZT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtMVwiPml0ZW0gb25lPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTJcIj5pdGVtIHR3bzwvZWwtbWVudS1pdGVtPlxuICAgICAgPC9lbC1tZW51LWl0ZW0tZ3JvdXA+XG4gICAgICA8ZWwtbWVudS1pdGVtLWdyb3VwIHRpdGxlPVwiR3JvdXAgVHdvXCI+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTNcIj5pdGVtIHRocmVlPC9lbC1tZW51LWl0ZW0+XG4gICAgICA8L2VsLW1lbnUtaXRlbS1ncm91cD5cbiAgICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjEtNFwiIDpleHBhbmQtY2xvc2UtaWNvbj1cIkNhcmV0Qm90dG9tXCIgOmV4cGFuZC1vcGVuLWljb249XCJDYXJldFRvcFwiXG4gICAgICAgIDpjb2xsYXBzZS1jbG9zZS1pY29uPVwiQ2FyZXRSaWdodFwiIDpjb2xsYXBzZS1vcGVuLWljb249XCJDYXJldExlZnRcIj5cbiAgICAgICAgPHRlbXBsYXRlICN0aXRsZT48c3Bhbj5pdGVtIGZvdXI8L3NwYW4+PC90ZW1wbGF0ZT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtNC0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgPC9lbC1zdWItbWVudT5cbiAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMlwiPlxuICAgICAgPGVsLWljb24+XG4gICAgICAgIDxpY29uLW1lbnUgLz5cbiAgICAgIDwvZWwtaWNvbj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+TmF2aWdhdG9yIFR3bzwvdGVtcGxhdGU+XG4gICAgPC9lbC1tZW51LWl0ZW0+XG4gICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjNcIiBkaXNhYmxlZD5cbiAgICAgIDxlbC1pY29uPlxuICAgICAgICA8ZG9jdW1lbnQgLz5cbiAgICAgIDwvZWwtaWNvbj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+TmF2aWdhdG9yIFRocmVlPC90ZW1wbGF0ZT5cbiAgICA8L2VsLW1lbnUtaXRlbT5cbiAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNFwiPlxuICAgICAgPGVsLWljb24+XG4gICAgICAgIDxzZXR0aW5nIC8+XG4gICAgICA8L2VsLWljb24+XG4gICAgICA8dGVtcGxhdGUgI3RpdGxlPk5hdmlnYXRvciBGb3VyPC90ZW1wbGF0ZT5cbiAgICA8L2VsLW1lbnUtaXRlbT5cbiAgPC9lbC1tZW51PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7XG4gIENhcmV0Qm90dG9tLFxuICBDYXJldExlZnQsXG4gIENhcmV0UmlnaHQsXG4gIENhcmV0VG9wLFxuICBEb2N1bWVudCxcbiAgTWVudSBhcyBJY29uTWVudSxcbiAgTG9jYXRpb24sXG4gIFNldHRpbmcsXG59IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBpc0NvbGxhcHNlID0gcmVmKGZhbHNlKVxuY29uc3QgaGFuZGxlT3BlbiA9IChrZXk6IHN0cmluZywga2V5UGF0aDogc3RyaW5nW10pID0+IHtcbiAgY29uc29sZS5sb2coa2V5LCBrZXlQYXRoKVxufVxuY29uc3QgaGFuZGxlQ2xvc2UgPSAoa2V5OiBzdHJpbmcsIGtleVBhdGg6IHN0cmluZ1tdKSA9PiB7XG4gIGNvbnNvbGUubG9nKGtleSwga2V5UGF0aClcbn1cbjwvc2NyaXB0PlxuXG48c3R5bGU+XG4uZWwtbWVudS12ZXJ0aWNhbC1kZW1vOm5vdCguZWwtbWVudS0tY29sbGFwc2UpIHtcbiAgd2lkdGg6IDIwMHB4O1xuICBtaW4taGVpZ2h0OiA0MDBweDtcbn1cbjwvc3R5bGU+IiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTk0NDQtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AMy4yLjM3L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEAzLjIuMzcvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy05NDQ0LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTk0NDQtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctOTQ0NC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

File: packages/components/menu/src/sub-menu.ts [hunk 1]
```diff
@@ -51,6 +57,22 @@ export const subMenuProps = buildProps({
     type: Number,
     default: 6,
   },
+  expandCloseIcon: {
+    type: iconPropType,
+    default: '',
+  },
```
Comment (holazz) [inline packages/components/menu/src/sub-menu.ts:63]: How about changing the type here to this?

```ts
expandCloseIcon: {
	type: definePropType<Component | (() => VNode)>([Object, Function]),
},
```

Same in other places.

Comment (xiaoxian521) [reply packages/components/menu/src/sub-menu.ts]: Done

Comment (holazz) [reply packages/components/menu/src/sub-menu.ts]: Nice!

File: packages/components/menu/src/sub-menu.ts [hunk 2]
```diff
@@ -51,6 +57,22 @@ export const subMenuProps = buildProps({
     type: Number,
     default: 6,
   },
+  expandCloseIcon: {
+    type: definePropType<Component | (() => VNode)>([Object, Function]),
+    default: '',
```
Comment (btea) [inline packages/components/menu/src/sub-menu.ts:62]: The default value doesn't seem to match the type.

File: packages/components/menu/src/sub-menu.ts [hunk 3]
```diff
@@ -92,7 +114,15 @@ export default defineComponent({
...
-        ? ArrowDown
+        ? props.expandCloseIcon && props.expandOpenIcon
+          ? opened.value
+            ? props.expandOpenIcon
+            : props.expandCloseIcon
+          : ArrowDown
+        : props.collapseCloseIcon && props.collapseOpenIcon
+        ? opened.value
+          ? props.collapseOpenIcon
+          : props.collapseCloseIcon
```
Comment (btea) [inline packages/components/menu/src/sub-menu.ts:125]: Could the logic be adjusted here? It doesn't look very intuitive.

File: packages/components/menu/src/sub-menu.ts (hunk 2 above)
Comment (xiaoxian521) [reply packages/components/menu/src/sub-menu.ts:62]: 没有默认值

File: packages/components/menu/src/sub-menu.ts (hunk 3 above)
Comment (xiaoxian521) [reply packages/components/menu/src/sub-menu.ts:125]: 这并没有任何影响

**Review by holazz — APPROVED** [review]

**Comment by btea** [comment]
@xiaoxian521 Maybe we can add an example to the documentation.

**Pull request merged**